src/Widgets/Popovers/AvatarPopover.vala

```diff
@@ -59,19 +59,22 @@ namespace SwitchboardPlugUserAccounts.Widgets {
         }
 
         private void select_from_file () {
-            var file_dialog = new Gtk.FileChooserDialog (_("Select an image"),
-            get_parent_window () as Gtk.Window?, Gtk.FileChooserAction.OPEN, _("Cancel"),
-            Gtk.ResponseType.CANCEL, _("Open"), Gtk.ResponseType.ACCEPT);
-
-            Gtk.FileFilter filter = new Gtk.FileFilter ();
+            var filter = new Gtk.FileFilter ();
             filter.set_filter_name (_("Images"));
-            file_dialog.set_filter (filter);
             filter.add_mime_type ("image/jpeg");
             filter.add_mime_type ("image/jpg");
             filter.add_mime_type ("image/png");
 
-            // Add a preview widget
-            Gtk.Image preview_area = new Gtk.Image ();
+            var preview_area = new Gtk.Image ();
+
+            var file_dialog = new Gtk.FileChooserNative (
+                _("Select an image"),
+                get_parent_window () as Gtk.Window?,
+                Gtk.FileChooserAction.OPEN,
+                _("Open"),
+                _("Cancel")
+            );
+            file_dialog.filter = filter;
             file_dialog.set_preview_widget (preview_area);
```

Comment:
As far as I know, this will not work on a true native filechooser (at the moment), only for the fallback GtkFileChooser.  You cannot send extra widgets over the Dbus portal except that you can add extra comboboxes containing choices by sending them in serialized form.  Maybe the Pantheon implementation will add that ability somehow??

Reply:
Yes, we’ll want to implement image previews as a part of our chooser instead of each app trying to reimplement custom preview widgets

Reply:
My understanding is that our filechooser will be activated via the DBus portal and `file_dialog.set_preview_widget (preview_area)` will not work. I assume you are leaving this in for now so current behaviour is maintained?

Reply:
Right. This only works with the fallback chooser. When the native filechooser lands, ideally this would be unnecessary anyways since the new chooser should have its own built-in preview pane